middleware/isInterestedUser.js

```diff
@@ -0,0 +1,27 @@
+const customError = require('../helpers/customErrorHandler');
+const Book = require('../models/Book');
+
+exports.isInterestedUser = async (req, res, next) => {
+  const { userId, bookId } = req.body;
+
+  if (!userId || !bookId) {
+    return next(customError('A user ID and a book ID must be provided', 400));
+  }
+
+  const book = await Book.findById(bookId);
+  console.log(book);
+
+  const interestedUserArray = book.interestedUsers.filter(
+    (id) => id.toString() === userId.toString()
+  );
```

Comment:
just as info : another way would be to use includes. it gives back true or false . 
`book.interestedUsers.includes(userId) ?... :....`

Reply:
true, i shall have to check its going to work with ids, because saying an id is like another id sometimes doesn't work but i shall check after i push my current branch :)

Reply:
i can review then again if you want